WordPress/src/main/java/org/wordpress/android/viewmodel/PluginBrowserViewModel.kt

```diff
@@ -0,0 +1,465 @@
+package org.wordpress.android.viewmodel
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import android.os.Bundle
+import android.os.Handler
+import android.support.annotation.WorkerThread
+import android.text.TextUtils
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.PluginActionBuilder
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.plugin.ImmutablePluginModel
+import org.wordpress.android.fluxc.model.plugin.PluginDirectoryType
+import org.wordpress.android.fluxc.store.PluginStore
+import org.wordpress.android.fluxc.store.PluginStore.FetchPluginDirectoryPayload
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T
+import org.wordpress.android.util.StringUtils
+import java.util.ArrayList
+import java.util.HashMap
+import java.util.HashSet
+import javax.inject.Inject
+import kotlin.properties.Delegates
+
+private const val KEY_SEARCH_QUERY = "KEY_SEARCH_QUERY"
```

Comment:
I am not sure whether a `companion object` or the `const` declaration is the way to go. `companion object` feels a bit unnecessary to me, but I am happy to follow the convention if there is any.

Reply:
This is fine, just not that if you'd like to change it in the future to public and access it from java you'll have to use `PluginBrowserViewModelKt.CONSTANT` format.